lib/potluck.rb

```diff
@@ -0,0 +1,49 @@
+require "pry"
+require "./lib/dish"
+
+class Potluck
+  attr_reader :date, :dishes
+  def initialize(date = "0-0-0")
+    @date = date
+    @dishes = []
+  end
+
+  def add_dish(dish)
+    @dishes << dish
+  end
+
+  def get_all_from_category(category)
+    category_dishes = @dishes.map do |dish|
+      if dish.category == category
+        dish
+      end
+    end
+    category_dishes.compact
+  end
+
+  def menu
+    hash = Hash.new{0}
+    x = @dishes.map do |dish|
+      dish.category
+    end
+
+    x.map do |category|
+      category_dishes = @dishes.map do |dish|
+        if dish.category == category
+          dish.name
+        end
+      end
+      category_dishes.compact
+      hash[category] = (category_dishes.compact).sort
+    end
+    hash
```

Comment:
Interesting! Like that you were able to power through this!

Assume that the main concern here is how to create keys and values. One thing to consider is iterating over the array of dishes, then doing a check to see if a key for that dish exists. If so, go ahead and add to it. If not, create the key value pair.